src/uint/boxed/shl.rs

```diff
@@ -111,19 +111,6 @@ impl BoxedUint {
         Some(())
     }
 
-    /// Computes `self << shift`.
-    /// Returns `None` if `shift >= self.bits_precision()`.
-    ///
-    /// NOTE: this operation is variable time with respect to `shift` *ONLY*.
-    ///
-    /// When used with a fixed `shift`, this function is constant-time with respect to `self`.
-    #[inline(always)]
-    pub fn shl_vartime(&self, shift: u32) -> Option<Self> {
-        let mut result = Self::zero_with_precision(self.bits_precision());
-        let success = self.shl_vartime_into(&mut result, shift);
-        success.map(|_| result)
-    }
```

Comment:
?

Reply:
I originally made them to accommodate the requirement from `crypto-primes`, but now I realized I can use `overflowing_shl`/`overflowing_shr` to achieve the same thing, so I felt no need to add `shl_vartime`/`shr_vartime` for `BoxedUint`.

Reply:
Err, I added `shl_vartime` in #330. The reason we have `*_vartime` equivalents of the `sh*` methods is because they can be quite a bit faster.

Reply:
Oh now I see the reason for all the replaced `_vartime` methods previously. Yes, we should keep those.